xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-picker/xwiki-platform-attachment-picker-webjar/src/main/webjar/attachmentGalleryPicker.js

```diff
@@ -179,16 +179,18 @@ define('xwiki-attachment-picker',
       initSelectedAttachment() {
         this.resultsBlock.on('click', 'a', (event) => {
           event.preventDefault();
-          const parent = $(event.currentTarget).parents('.attachmentGroup');
+          if (event.detail === 1) {
```

Comment:
Same here, I'm not sure to understand why the rest of this method needs to be wrapped in this if.

Reply:
I added more explanation in https://github.com/xwiki/xwiki-platform/pull/2817/commits/2368ef30d1e20bedcae42b80bc8f4fc5408fffd2.
The issue I was facing was that a click toggles selection of the image.
When double clicking, it created unexpected behaviour where the image would be selected only in between the two clicks. With this condition, we make sure we select (or unselect) only on the first click.

We can use the detail attribute of the click event to get the `number` of the click in the clickchain, see [this paragraph about it on MDN Webdocs](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#usage_notes).

Reply:
Apparently, [click events](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) are also sent when pressing space or enter while the element is focused. Did you check that this change doesn't interfere with this?